apps/OboeTester/app/src/main/java/com/mobileer/oboetester/IntentBasedTestSupport.java

```diff
@@ -173,65 +174,67 @@ public static int getVolumeStreamTypeFromBundle(Bundle bundle) {
             return StreamConfiguration.UNSPECIFIED;
-        }
-        switch (channelMaskText) {
-            case VALUE_CHANNEL_MONO:
-                return StreamConfiguration.CHANNEL_MONO;
-            case VALUE_CHANNEL_STEREO:
-                return StreamConfiguration.CHANNEL_STEREO;
-            case VALUE_CHANNEL_2POINT1:
-                return StreamConfiguration.CHANNEL_2POINT1;
-            case VALUE_CHANNEL_TRI:
-                return StreamConfiguration.CHANNEL_TRI;
-            case VALUE_CHANNEL_TRI_BACK:
-                return StreamConfiguration.CHANNEL_TRI_BACK;
-            case VALUE_CHANNEL_3POINT1:
-                return StreamConfiguration.CHANNEL_3POINT1;
-            case VALUE_CHANNEL_2POINT0POINT2:
-                return StreamConfiguration.CHANNEL_2POINT0POINT2;
-            case VALUE_CHANNEL_2POINT1POINT2:
-                return StreamConfiguration.CHANNEL_2POINT1POINT2;
-            case VALUE_CHANNEL_3POINT0POINT2:
-                return StreamConfiguration.CHANNEL_3POINT0POINT2;
-            case VALUE_CHANNEL_3POINT1POINT2:
-                return StreamConfiguration.CHANNEL_3POINT1POINT2;
-            case VALUE_CHANNEL_QUAD:
-                return StreamConfiguration.CHANNEL_QUAD;
-            case VALUE_CHANNEL_QUAD_SIDE:
-                return StreamConfiguration.CHANNEL_QUAD_SIDE;
-            case VALUE_CHANNEL_SURROUND:
-                return StreamConfiguration.CHANNEL_SURROUND;
-            case VALUE_CHANNEL_PENTA:
-                return StreamConfiguration.CHANNEL_PENTA;
-            case VALUE_CHANNEL_5POINT1:
-                return StreamConfiguration.CHANNEL_5POINT1;
-            case VALUE_CHANNEL_5POINT1_SIDE:
-                return StreamConfiguration.CHANNEL_5POINT1_SIDE;
-            case VALUE_CHANNEL_6POINT1:
-                return StreamConfiguration.CHANNEL_6POINT1;
-            case VALUE_CHANNEL_7POINT1:
-                return StreamConfiguration.CHANNEL_7POINT1;
-            case VALUE_CHANNEL_5POINT1POINT2:
-                return StreamConfiguration.CHANNEL_5POINT1POINT2;
-            case VALUE_CHANNEL_5POINT1POINT4:
-                return StreamConfiguration.CHANNEL_5POINT1POINT4;
-            case VALUE_CHANNEL_7POINT1POINT2:
-                return StreamConfiguration.CHANNEL_7POINT1POINT2;
-            case VALUE_CHANNEL_7POINT1POINT4:
-                return StreamConfiguration.CHANNEL_7POINT1POINT4;
-            case VALUE_CHANNEL_9POINT1POINT4:
-                return StreamConfiguration.CHANNEL_9POINT1POINT4;
-            case VALUE_CHANNEL_9POINT1POINT6:
-                return StreamConfiguration.CHANNEL_9POINT1POINT6;
-            case VALUE_CHANNEL_FRONT_BACK:
-                return StreamConfiguration.CHANNEL_FRONT_BACK;
-            default:
-                throw new IllegalArgumentException(
-                        KEY_CHANNEL_MASK + " invalid: " + channelMaskText);
+        } else if (channelMaskText.equals(VALUE_CHANNEL_MONO)) {
```

Comment:
Seems switch case will be more efficient. See https://docs.oracle.com/javase/7/docs/technotes/guides/language/strings-switch.html.

Reply:
Done in Commit 3.